imap_processing/cdf/cdf_attribute_manager.py

```diff
@@ -0,0 +1,207 @@
+        ----------
+        data_dir: Path
+            The directory containing the schema and variable files - nominally config/
+        """
+        # TODO: Split up schema source and data source?
+        self.source_dir = data_dir
+
+        # TODO: copied from hermes_core. Currently we can use default schema, but
+        # We should add some way of extending the schema and remove all the HERMES
+        # specific stuff
+        # Data Validation, Complaiance,
+        self.global_attribute_schema = self._load_default_global_attr_schema()
+
+        # Data Validation and Compliance for Variable Data
+        self.variable_attribute_schema = self._load_default_variable_attr_schema()
+
+        # Load Default IMAP Global Attributes
+        self.global_attributes = self._load_yaml_data(
+            self.source_dir / DEFAULT_GLOBAL_CDF_ATTRS_FILE
+        )
+        self.variable_attributes = dict()
+
+    def _load_default_global_attr_schema(self) -> dict:
+        """
+        Load the default global schema from the source directory.
+
+        Returns
+        -------
+        the dict representing the global schema
+        """
+        default_schema_path = (
+            self.source_dir / "shared" / DEFAULT_GLOBAL_CDF_ATTRS_SCHEMA_FILE
+        )
+        # Load the Schema
+        return self._load_yaml_data(default_schema_path)
+
+    def _load_default_variable_attr_schema(self) -> dict:
+        """
+        Load the default variable schema from the source directory.
+
+        Returns
+        -------
+        the dict representing the variable schema
+        """
+        default_schema_path = (
+            self.source_dir / "shared" / DEFAULT_VARIABLE_CDF_ATTRS_SCHEMA_FILE
+        )
+        # Load the Schema
+        return CdfAttributeManager._load_yaml_data(default_schema_path)
+
+    def load_global_attributes(self, file_path: str):
+        """
+        Update the global attributes property with the attributes from the file.
+
+        Parameters
+        ----------
+        file_path: str
+            file path to load, under self.source_dir.
+
+        """
+        self.global_attributes.update(self._load_yaml_data(self.source_dir / file_path))
+
+    @staticmethod
+    def _load_yaml_data(file_path: str | Path) -> dict:
+        """
+        Load a yaml file from the provided path.
+
+        Parameters
+        ----------
+        file_path: str | Path
+            path to the yaml file to load.
+
+        Returns
+        -------
+            Loaded yaml.
+        """
+        with open(file_path) as file:
+            return yaml.safe_load(file)
+
+    def get_global_attributes(self, instrument_id=None) -> dict:
+        """
+        Generate a dictionary global attributes based off the loaded schema and attrs.
+
+        Validates against the global schema to ensure all required variables are
+        present. It can also include instrument specific global attributes if
+        instrumet_id is set.
```

Comment:
typo + We should highlight that instrument_id is a quantity. I don't remember if we are bolding these are double-backticking them.
```suggestion
        **instrument_id** is set.
```